sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Keyvault Secret definition
+ */
+
+#pragma once
+#include <azure/keyvault/secrets/keyvault_secret_properties.hpp>
+
+#include <unordered_map>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
+  struct KeyVaultSecret final
+  {
+    /**
+     * @brief The secret value.
+     *
+     */
+    std::string Value;
+
+    /**
+     * @brief The secret id.
+     *
+     */
+    std::string Id;
+
+    /**
+     * @brief The secret Properties bundle.
+     *
+     */
+    KeyvaultSecretProperties Properties;
+
+    /**
+     * @brief The content type of the secret.
+     *
+     */
+    Azure::Nullable<std::string> ContentType;
+
+    /**
+     * @brief  If this is a secret backing a KV certificate, then this field specifies the
+     * corresponding key backing the KV certificate.
+     *
+     */
+    Azure::Nullable<std::string> KeyId;
+
+    /**
+     * @brief Application specific metadata in the form of key-value pairs.
+     *
+     */
+    std::unordered_map<std::string, std::string> Tags;
+
+    /**
+     * @brief True if the secret's lifetime is managed by key vault. If this is a secret
+     * backing a certificate, then managed will be true.
+     *
+     */
+    bool Managed = false;
```

Comment:
These should be defined in `SecretProperties` to be congruent with other languages. See https://docs.microsoft.com/en-us/dotnet/api/azure.security.keyvault.secrets.secretproperties?view=azure-dotnet for an example.

Reply:
done